metal/src/kernels/matmul/mod.rs

```diff
@@ -636,36 +717,51 @@ mod tests {
         F: Datum + Float,
         usize: AsPrimitive<F>,
     {
-        type Parameters = ();
+        type Parameters = MmmProblemParams;
         type Strategy = BoxedStrategy<Self>;
 
-        fn arbitrary_with(_: ()) -> Self::Strategy {
-            (1usize..10, 1usize..20, 1usize..20, 1usize..20)
-                .prop_flat_map(|(b, m, k, n)| {
-                    let lhs_len = b * m * k;
-                    let rhs_len = b * k * n;
-                    let datum = (0usize..10).prop_map(|x| x.as_());
+        fn arbitrary_with(params: MmmProblemParams) -> Self::Strategy {
+            (1usize..4, 1usize..128, 1usize..256, 1usize..128)
+                .prop_flat_map(move |(b, m, mut k, n)| {
+                    if params.q4_0_weights {
+                        k = k.div_ceil(32) * 32
+                    };
+
+                    let mut rng = rand::thread_rng();
+                    let lhs_data: Vec<F> = (0..b * m * k)  // Create a vector with 10 elements
+                    .map(|_| F::from(rng.gen_range(0.0..1.0)).unwrap()) // Generate a random float in [0.0, 1.0)
```

Comment:
mmm... this is not good. we should not rand() ourselves, but let the proptest strategies do it in predictable and simplifiable manner. The idea is to use `proptest::collections::vec()` or something like that.